packages/flutter_tools/gradle/flutter.gradle

```diff
@@ -634,6 +634,38 @@ class FlutterPlugin implements Plugin<Project> {
                     }
                 }
             }
+            // Copy the output APKs into a known location, so `flutter run` or `flutter build apk`
+            // can discover them. By default, this is `<app-dir>/build/app/outputs/flutter-apk/<filename>.apk`.
+            //
+            // The filename consists of `app<-abi>?<-flavor-name>?-<build-mode>.apk`.
+            // Where:
+            //   * `abi` can be `armeabi-v7a|arm64-v8a|x86|x86_64` only if the flag `split-per-abi` is set.
+            //   * `flavor-name` is the flavor used to build the app in lower case if the assemble task is called.
+            //   * `build-mode` can be `release|debug|profile`.
+            variant.outputs.all { output ->
+                variant.assembleProvider.get().doLast {
+                    def outputDirectory = variant.packageApplicationProvider.get().outputDirectory
+                    //  `outputDirectory` is a `DirectoryProperty` in AGP 4.1.
+                    String outputDirectoryStr = outputDirectory.metaClass.respondsTo(outputDirectory, "get")
+                        ? outputDirectory.get() : outputDirectory
+                    String filename = "app"
+                    String abi = output.getFilter(OutputFile.ABI)
+                    if (abi != null && !abi.isEmpty()) {
+                        filename += "-${abi}"
+                    }
+                    if (variant.flavorName != null && !variant.flavorName.isEmpty()) {
```

Comment:
What does this do for multiple flavor dimensions? Do we get `dim1Dim2`?

Reply:
Yup. I tested with multiple flavors and it works as expected